### PR TITLE
 🌱 Introduce Kubeadm v1beta3

### LINKED ISSUE #4445: Implement support for kubeadm v1beta3 types
**User Story**

As a user I would like CAPI to use latest kubeadm API types when possible

**Detailed Description**

kubeadm is going to introduce v1beta3 API types during the v1.22 release cycle, and at the same time, drop v1beta1 types.

CAPI should follow these changes.

**Anything else you would like to add:**

Feasibility of dropping support for v1beta1 types should be evaluated according to the outcomes of the discussion on https://github.com/kubernetes-sigs/cluster-api/issues/4444 

/kind feature

### PR DESCRIPTION
**Pull request by shysank**
**What this PR does / why we need it**:

With v1beta3 merged in Kubeadm, this pr introduces v1beta3 to bootstrap/kubeadm types with all the scaffolding.  

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #4445

### CODE REVIEW

**Comment by vincepri** [comment]
LGTM apart from the linter errors and verification job

/assign @fabriziopandini @CecileRobertMichon

File: bootstrap/kubeadm/types/utils.go [hunk 1]
```diff
@@ -63,14 +69,17 @@ func KubeVersionToKubeadmAPIGroupVersion(version semver.Version) (schema.GroupVe
 	case version.LT(v1beta2KubeadmVersion):
 		// NOTE: All the Kubernetes version >= v1.13 and < v1.15 should use the kubeadm API version v1beta1
 		return v1beta1.GroupVersion, nil
+		// NOTE: All the Kubernetes version >= v1.15 and < v1.22 should use the kubeadm API version v1beta1
+	case version.LT(v1beta3KubeadmVersion):
```
Comment (fabriziopandini) [inline bootstrap/kubeadm/types/utils.go:73]:
```suggestion
	case version.LT(v1beta3KubeadmVersion):
		// NOTE: All the Kubernetes version >= v1.15 and < v1.22 should use the kubeadm API version v1beta2
```

File: bootstrap/kubeadm/types/utils_test.go [hunk 2]
```diff
@@ -78,12 +79,20 @@ func TestKubeVersionToKubeadmAPIGroupVersion(t *testing.T) {
 			want:    v1beta2.GroupVersion,
 			wantErr: false,
 		},
+		{
+			name: "pass with kubernetes version for kubeadm API v1beta3",
```
Comment (fabriziopandini) [inline bootstrap/kubeadm/types/utils_test.go:83]: Let's also add a test "pass with minimum kubernetes version for kubeadm API v1beta3" (v1.22.0),

**Review by fabriziopandini — COMMENTED** [review]
lgtm apart two nits

**Comment by CecileRobertMichon** [comment]
Do we have a k8s 1.22 (or k8s HEAD) test running as part of this PR? It'd be good if we have a way to run k8s main conformance to validate changes like these similar to the ones we have in CAPA/CAPZ/soon CAPO

**Comment by neolit123** [comment]
> Fixes #4445

having the v1beta3/types.go here as a placeholder seems fine, but should it be updated once kubeadm work on v1beta3 finishes with the 1.22 release - i.e. are fields are added / removed?

**Comment by fabriziopandini** [comment]
> Do we have a k8s 1.22 (or k8s HEAD) test running as part of this PR?

Yes, it is one of our PR optional jobs (in addition to a periodic periodic job).

/test pull-cluster-api-e2e-workload-upgrade-1-21-latest-main

> having the v1beta3 here as a placeholder seems fine, but should it be updated once kubeadm work on v1beta3 finishes with the 1.22 release - i.e. fields are added / removed?

Of course, but without this our tests are failing

**Review by vincepri — APPROVED** [review]
/approve
/kind release-blocking
